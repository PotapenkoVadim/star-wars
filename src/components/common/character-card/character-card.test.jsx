import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import CharacterCard from './index';

const character = {
  birth_year: '19BBY',
  eye_color: 'blue',
  gender: 'male',
  hair_color: 'blond',
  name: 'Luke Skywalker',
  skin_color: 'fair'
};

const handleClick = jest.fn();

describe('Character card', () => {
  test('Renders the information correctly', () => {
    render(<CharacterCard character={character} handleClick={handleClick} />);

    expect(screen.getByText(/luke skywalker/i)).toBeInTheDocument();
    expect(screen.getByText(/19BBY/i)).toBeInTheDocument();

    expect(screen.queryByText(/C-3PO/i)).toBeNull();
    expect(screen.queryByText(/unknown/i)).toBeNull();
    expect(screen.queryByText(/n\/a/i)).toBeNull();
  });

  test('Correct operation of onClick', () => {
    render(<CharacterCard character={character} handleClick={handleClick} />);

    userEvent.click(screen.getByTestId('charactercard'));
    handleClick.mockReturnValue(character);

    expect(handleClick).toHaveBeenCalled();
    expect(handleClick()).toBe(character);
  });

  test('Match snapshot', () => {
    const { container } = render(<CharacterCard character={character} handleClick={handleClick} />);

    expect(container).toMatchSnapshot();
  });
});