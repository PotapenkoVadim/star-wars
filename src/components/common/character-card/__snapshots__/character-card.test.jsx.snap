// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character card Match snapshot 1`] = `
<div>
  <div
    class="character"
    data-testid="charactercard"
  >
    <div
      class="character__name"
    >
      Luke Skywalker
    </div>
    <div
      class="character__meta"
    />
    <div
      class="character__tags"
    >
      <span
        class="
      marker
      marker_brith
    "
      >
        19BBY
      </span>
      <span
        class="
      marker
      marker_male
    "
      >
        male
      </span>
    </div>
  </div>
</div>
`;
